src/Validator/Constraints/BonusPointsApplyValidator.php

```diff
@@ -17,16 +17,16 @@
 final class BonusPointsApplyValidator extends ConstraintValidator
 {
     /** @var BonusPointsStrategyRepositoryInterface */
-    private $bonusPointsResolver;
+    private $bonusPointsStrategyRepository;
 
     public function __construct(BonusPointsStrategyRepositoryInterface $bonusPointsStrategyRepository)
     {
-        $this->bonusPointsResolver = $bonusPointsStrategyRepository;
+        $this->bonusPointsStrategyRepository = $bonusPointsStrategyRepository;
     }
 
     public function validate($bonusPoints, Constraint $constraint): void
     {
-        $bonusPointsStrategies = $this->bonusPointsResolver->findAll();
+        $bonusPointsStrategies = $this->bonusPointsStrategyRepository->findAll();
 
         /** @var BonusPointsStrategyInterface $bonusPointsStrategy */
         foreach ($bonusPointsStrategies as $bonusPointsStrategy) {
```

Comment:
Find only active bonus points strategy and check if a given strategy can be used for this order.

`$this->bonusPointsStrategyRepository->findAllActive()`

`$this->bonusPointsStrategyEligibilityChecker->isEligible()`